appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/jmx/MBeanMetadataHelper.java

```diff
@@ -308,14 +313,21 @@ private MBeanMetadata createMetadata(
                         subAttribute
                 ),
                 formatMetadata(
-                        nonNull(metadata.getDescription()) ? metadata.getDescription() : metadata.getName(),
+                        metadata.getDescription().isPresent() ? metadata.getDescription().get() : metadata.getName(),
                         key,
                         attribute,
                         subAttribute
                 ),
                 metadata.getTypeRaw(),
-                metadata.getUnit()
+                metadata.getUnit().orElse(null)
         );
+        for (fish.payara.microprofile.metrics.jmx.Tag oldTag: metadata.getTags()) {
+            fish.payara.microprofile.metrics.jmx.Tag newTag = new fish.payara.microprofile.metrics.jmx.Tag();
```

Comment:
Constructor with name and value or package-access method `transform` could spare us from these shadowed types.

It could read like 
```
  metadata.getTags().stream().map(
    oldTag -> oldTag.transform(
        name -> formatMetadata(name, key, attribute, subattribute),
        value -> formatMetadata(value, key, attribute, subAttribute)
    )).foreach(newMetadata.getTags()::add)
````